### PR TITLE
Add intra-cluster test to subctl throughput benchmark

### PR DESCRIPTION
**Pull request by maayanf24**
Fixes issue #707

Signed-off-by: Maayan Friedman <maafried@redhat.com>

### CODE REVIEW

File: pkg/subctl/benchmark/throughput.go [hunk 1]
```diff
@@ -21,13 +21,35 @@ func StartThroughputTests() {
...
-	runThroughputTest(f, framework.GatewayNode)
+	if !intraCluster {
+		testParams := benchmarkTestParams{
+			ClusterA:            framework.ClusterA,
+			ClusterB:            framework.ClusterB,
+			ServerPodScheduling: framework.GatewayNode,
+			ClientPodScheduling: framework.GatewayNode,
+		}
+
+		framework.By("Performing throughput tests from Gateway pod to Gateway pod")
```
Comment (tpantelis) [inline pkg/subctl/benchmark/throughput.go:34]:
```suggestion
		framework.By(fmt.Sprintf("Performing throughput tests from Gateway pod on cluster %q to Gateway pod on cluster %q ", clusterAName, clusterBName))
```

File: pkg/subctl/benchmark/throughput.go [hunk 2]
```diff
@@ -21,13 +21,35 @@ func StartThroughputTests() {
...
+		runThroughputTest(f, testParams)
+
+		testParams.ServerPodScheduling = framework.NonGatewayNode
+		testParams.ClientPodScheduling = framework.NonGatewayNode
+		framework.By("Performing throughput tests from Non-Gateway pod to Non-Gateway pod")
```
Comment (tpantelis) [inline pkg/subctl/benchmark/throughput.go:39]:
```suggestion
                framework.By(fmt.Sprintf("Performing throughput tests from Non-Gateway pod on cluster %q to Non-Gateway pod on cluster %q ", clusterAName, clusterBName))
```

File: pkg/subctl/cmd/benchmark.go [hunk 3]
```diff
@@ -16,7 +16,7 @@ var (
 		Long:  "This command used to run various benchmark tests between two clusters",
 	}
 	benchmarkThroughputCmd = &cobra.Command{
-		Use:   "throughput <kubeconfig1> <kubeconfig2>",
+		Use:   "throughput <kubeconfig1> [<kubeconfig2>]",
 		Short: "Benchmark throughput between two clusters",
```
Comment (tpantelis) [inline pkg/subctl/cmd/benchmark.go:20]:
```suggestion
		Short: "Benchmark throughput",
```

File: pkg/subctl/cmd/benchmark.go [hunk 4]
```diff
@@ -16,7 +16,7 @@ var (
...
 		Long:  "This command runs throughput tests between two clusters",
```
Comment (tpantelis) [inline pkg/subctl/cmd/benchmark.go:21]:
```suggestion
		Long:  "This command runs throughput tests within a cluster or between two clusters",
```

File: pkg/subctl/benchmark/throughput.go [hunk 5]
```diff
@@ -21,13 +21,35 @@ func StartThroughputTests() {
...
 	})
 
-	f = initFramework("latency")
+	f = initFramework("throughput")
 
-	framework.By("Performing throughput tests from Gateway pod to Gateway pod")
-	runThroughputTest(f, framework.GatewayNode)
+	if !intraCluster {
+		testParams := benchmarkTestParams{
+			ClusterA:            framework.ClusterA,
```
Comment (tpantelis) [inline pkg/subctl/benchmark/throughput.go:28]:
```suggestion
			ClientCluster:            framework.ClusterA,
```

File: pkg/subctl/benchmark/throughput.go [hunk 6]
```diff
@@ -21,13 +21,35 @@ func StartThroughputTests() {
...
+			ClusterB:            framework.ClusterB,
```
Comment (tpantelis) [inline pkg/subctl/benchmark/throughput.go:29]:
```suggestion
			ServerCluster:            framework.ClusterB,
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

**Review by mangelajo — COMMENTED** [review]
Seems good in general, I like tpantelis suggestions, please ping me once handled.

**Review by mangelajo — APPROVED** [review]
Looks great now, thanks

**Review by sridhargaddam — APPROVED** [review]
LGTM, thanks @maayanf24

**Pull request merged**